src/swish/http.ss

```diff
@@ -161,17 +161,17 @@
           state
           (let ([state (load-watchers state)])
             ($state copy [mime-types (read-mime-types)]))))
-    (define (url->abs-paths path)
-      (map
-       (lambda (x) (path-combine (web-dir) x))
-       (cond
-        [(char=? (string-ref path (- (string-length path) 1)) #\/)
-         (list (string-append path "index.ss"))]
-        [(string=? (path-extension path) "")
-         (list path
-           (string-append path ".ss")
-           (path-combine path "index.ss"))]
-        [else (list path)])))
+    (define (url->abs-paths path) ;; path starts with "/"
+      (let ([path (path-combine (web-dir)
```

Comment:
Rebinding path changes the behavior on Windows. (web-dir) is a native OS path, ending in \ on Windows.  If the path argument is "/", then this new path binding now ends in "\".  The original code stopped on the first cond clause, while the new code stops on the second clause -- thus yielding 3 proposed abs-paths, rather than 1.

Reply:
Good catch! Maybe the first cond case should be (directory-separator? (string-ref ...)). That should fix it on Windows.

Reply:
It seems that the original path variable is in URL format, making "/" the only valid character. The newly introduced path variable could be a path in Windows, Unix, or mixed format.  I'm not sure directory-separator? is sufficient. We may need to test both "/" and "\".

Reply:
Yes, the new path variable is formed by path-combine from the platform-specific path and the URL path in posix format. As a result, it cannot be the empty string (because of the way path-combine works). On Windows systems, directory-separator? checks for / and \\, which works well here. On posix systems, directory-separator? checks for / only. In all cases, it does what we want here.

Reply:
I didn't realize directory-separator? considered both / and \ on Windows.  Let's try that and make sure we get the right number of results from attempting the root '/" url.

Reply:
I made the change, and it works on the root '/' url.

Reply:
I also used string-append instead of path-combine since the directory-separator? test immediately preceding is exactly what path-combine does.